tree/treeplayer/inc/ROOT/TVec.hxx

```diff
@@ -0,0 +1,381 @@
+// Author: Enrico Guiraud, Enric Tejedor, Danilo Piparo CERN  01/2018
+
+/*************************************************************************
+ * Copyright (C) 1995-2018, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#ifndef ROOT_TVEC
+#define ROOT_TVEC
+
+#include <ROOT/TVecAllocator.hxx>
+#include <ROOT/TypeTraits.hxx>
+
+#include <algorithm>
+#include <cmath>   // for sqrt
+#include <numeric> // for inner_product
+#include <sstream>
+#include <vector>
+
+namespace ROOT {
+
+namespace Experimental {
+
+namespace VecOps {
+
+template<typename T>
+using TCallTraits = typename ROOT::TypeTraits::CallableTraits<T>;
+
+template <typename T>
+using TVec = std::vector<T, ROOT::Detail::VecOps::TVecAllocator<T>>;
+
+} // End of Experimental NS
+
+} // End of VecOps NS
+
+// Other helpers
+namespace Internal {
+
+namespace VecOps {
+
+using namespace ROOT::Experimental::VecOps;
+
+void CheckSizes(size_t s0, size_t s1, const char *opName)
+{
+   if (s0 != s1) {
+      std::stringstream err;
+      err << "Cannot perform operation " << opName
+          << ". The array sizes differ: "
+          << s0 << " and " << s1 << std::endl;
+      throw std::runtime_error(err.str());
+   }
+}
+
+template <typename T, typename V, typename F>
+TVec<typename TCallTraits<F>::ret_type> Operate(const TVec<T> &v0, const TVec<V> &v1, const char* opName, F f)
+{
+   CheckSizes(v0.size(), v1.size(), opName);
+   TVec<typename TCallTraits<F>::ret_type> w;
+   w.resize(v0.size());
+   std::transform(v0.begin(), v0.end(), v1.begin(), w.begin(), f);
+   return w;
+}
+
+template <typename T, typename F>
+TVec<typename TCallTraits<F>::ret_type> Operate(const TVec<T> &v0, F f)
```

Comment:
wow this is `TDF::Map` implemented in #1612 !

Reply:
I apologise for having forgotten this.

Reply:
Operate is a terrible name to use instead of the widely understood *Map*